### ISSUE TITLE
No configure script in package

### ISSUE THREAD

**Issue by kssummer** [FastCGI-Archives/fcgi2#8]
From the README on how to build:

Unix:
```
    ./configure
    make
    make install
```

No configure script exists in the package. (I've downloaded the latest release v2.4.1)

**Comment by mcarbonneaux** [comment]
Try to execute autogen.sh before to generate configure from configure.in file.

**Comment by tehnick** [comment]
I have not looked at build scripts yet, but quick test gives such results for now:

```
$ git status
On branch fix-scripts
nothing to commit, working tree clean

$ ./autogen.sh 
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.in,
libtoolize: and rerunning libtoolize and aclocal.
libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
configure.in:8: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.  For more info, see:
configure.in:8: https://www.gnu.org/software/automake/manual/automake.html#Modernize-AM_005fINIT_005fAUTOMAKE-invocation
configure.in:12: installing './compile'
configure.in:8: installing './missing'
Makefile.am: installing './INSTALL'
Makefile.am: error: required file './NEWS' not found
Makefile.am: error: required file './README' not found
Makefile.am: error: required file './AUTHORS' not found
Makefile.am: error: required file './ChangeLog' not found
Makefile.am: installing './COPYING' using GNU General Public License v3 file
Makefile.am:     Consider adding the COPYING file to the version control system
Makefile.am:     for your code, to avoid questions about which license your project uses
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
cgi-fcgi/Makefile.am: installing './depcomp'
cgi-fcgi/Makefile.am:5: warning: 'CPPFLAGS' is a user variable, you should not override it;
cgi-fcgi/Makefile.am:5: use 'AM_CPPFLAGS' instead
examples/Makefile.am:14: warning: 'CPPFLAGS' is a user variable, you should not override it;
examples/Makefile.am:14: use 'AM_CPPFLAGS' instead
libfcgi/Makefile.am:4: warning: 'CPPFLAGS' is a user variable, you should not override it;
libfcgi/Makefile.am:4: use 'AM_CPPFLAGS' instead
autoreconf: automake failed with exit status: 1

$ git status
On branch fix-scripts
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   compile
        modified:   config.guess
        modified:   config.sub
        modified:   depcomp
        modified:   install-sh
        modified:   ltmain.sh
        modified:   missing
        modified:   mkinstalldirs

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        COPYING
        INSTALL
        aclocal.m4
        autom4te.cache/
        configure
        fcgi_config.h.in

no changes added to commit (use "git add" and/or "git commit -a")

$ ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... mt
checking if mt is a manifest tool... no
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether g++ works... yes
checking whether cin has a streambuf assignment operator... no
checking whether char_type is defined in the context of streambuf... yes
checking for gethostbyname in -lnsl... yes
checking for socket in -lsocket... no
checking pthread.h usability... yes
checking pthread.h presence... yes
checking for pthread.h... yes
checking for the pthreads library -lpthreads... no
checking whether pthreads work without any flags... no
checking whether pthreads work with -Kthread... no
checking whether pthreads work with -kthread... no
checking for the pthreads library -llthread... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking if more special flags are required for pthreads... no
checking for cc_r... gcc
checking for ssize_t... yes
checking for sun_len in sys/un.h... no
checking for fpos_t in stdio.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking netdb.h usability... yes
checking netdb.h presence... yes
checking for netdb.h... yes
checking netinet/in.h usability... yes
checking netinet/in.h presence... yes
checking for netinet/in.h... yes
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking for unistd.h... (cached) yes
checking for a fileno() prototype in stdio.h... yes
checking whether cross-process locking is required by accept()... no
checking whether va_arg(arg, long double) crashes the compiler... no
checking for an ANSI C-conforming const... yes
checking for strerror... yes
checking for inline... inline
checking for C compiler warning flags... -Wall
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: error: cannot find input file: `Makefile.in'
```

**Comment by tehnick** [comment]
Hmm, I see https://github.com/FastCGI-Archives/fcgi2/pull/6 . It should fix a problem. I'll check it a bit later.

**Comment by mcarbonneaux** [comment]
i've merger the pool request.
you can retry ?

**Comment by mcarbonneaux** [comment]
i've added circleci build! and it's build fine on debian:latest.

**Comment by tehnick** [comment]
@mcarbonneaux 
Hi. I may confirm that now library could be built without issues. Thank you.

**Comment by tehnick** [comment]
Hmm, unfortunately these build scripts are still not perfect.

For example:
1) Build and installation outside of sources tree fails:
```
...
make[1]: Entering directory `/home/boradmin/Hobby/Devel/FastCGI/build-fcgi2__my/Ubuntu-14.04_i386_shared/libfcgi'
make[2]: Entering directory `/home/boradmin/Hobby/Devel/FastCGI/build-fcgi2__my/Ubuntu-14.04_i386_shared/libfcgi'
 /bin/mkdir -p '../Ubuntu-14.04_i386_shared-out/usr/lib'
 /bin/bash ../libtool   --mode=install /usr/bin/install -c   libfcgi.la libfcgi++.la '../Ubuntu-14.04_i386_shared-out/usr/lib'
Usage: /home/boradmin/Hobby/Devel/FastCGI/build-fcgi2__my/Ubuntu-14.04_i386_shared/libtool [OPTION]... [MODE-ARG]...
Try 'libtool --help' for more information.
libtool:   error: '../Ubuntu-14.04_i386_shared-out/usr/lib' must be an absolute directory name
make[2]: *** [install-libLTLIBRARIES] Error 1
make[2]: Target `install-exec-am' not remade because of errors.
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/home/boradmin/Hobby/Devel/FastCGI/build-fcgi2__my/Ubuntu-14.04_i386_shared/libfcgi'
make[1]: *** [install-am] Error 2
make[1]: Target `install' not remade because of errors.
...
```
2) Cross-compilation using MinGW failed, because MinGW is not detected properly:
```
...
libtool: compile:  i686-w64-mingw32.shared-g++ -DHAVE_CONFIG_H -I. -I/home/boradmin/Hobby/Devel/FastCGI/fcgi2__my/libfcgi -I.. -I/home/boradmin/Hobby/Devel/FastCGI/fcgi2__my/include -g -O2 -MT fcgio.lo -MD -MP -MF .deps/fcgio.Tpo -c /home/boradmin/Hobby/Devel/FastCGI/fcgi2__my/libfcgi/fcgio.cpp  -DDLL_EXPORT -DPIC -o .libs/fcgio.o
/home/boradmin/Hobby/Devel/FastCGI/fcgi2__my/libfcgi/os_unix.c:27:23: fatal error: arpa/inet.h: No such file or directory
 #include <arpa/inet.h>
                       ^
compilation terminated.
make[2]: *** [Makefile:515: libfcgi_la-os_unix.lo] Error 1
...
/home/boradmin/Hobby/Devel/FastCGI/fcgi2__my/examples/log-dump.c: In function 'main':
/home/boradmin/Hobby/Devel/FastCGI/fcgi2__my/examples/log-dump.c:54:6: warning: implicit declaration of function 'kill' [-Wimplicit-function-declaration]
      kill(getpid(), SIGQUIT);
      ^~~~
/home/boradmin/Hobby/Devel/FastCGI/fcgi2__my/examples/log-dump.c:54:21: error: 'SIGQUIT' undeclared (first use in this function)
      kill(getpid(), SIGQUIT);
                     ^~~~~~~
/home/boradmin/Hobby/Devel/FastCGI/fcgi2__my/examples/log-dump.c:54:21: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [Makefile:472: log-dump.o] Error 1
...
```

I'll look at these problems more deeply later and will open new bug reports or pull requests for them.

**Comment by tehnick** [comment]
@mcarbonneaux 
1. I have forgot that this is the issue (installation using relative path to directory) with all autotools based projects, so it not a problem in `fcgi2`.
2. Fixed in #9.
